packages/analytics/src/tests/analytics.test.ts

```diff
@@ -0,0 +1,95 @@
+import fixtures from './fixtures/encodeDataToQueryString';
+import { encodeDataToQueryString, getRandomId } from '../utils';
+
+import { Analytics } from '../analytics';
+
+describe('analytics', () => {
+    it('should report if enabled and do not report when disabled', () => {
+        const mockFetchPromise = Promise.resolve({
+            json: () => Promise.resolve({}),
+        });
+        global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);
+        // @ts-ignore
+        jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);
+
+        const timestamp = new Date().getTime();
+        jest.spyOn(Date, 'now').mockImplementation(() => timestamp);
+
+        const app = 'suite';
+        const environment = 'desktop';
+        const isDev = false;
+        const instanceId = getRandomId();
+        const sessionId = getRandomId();
+        const commitId = 'abc';
+
+        const analytics = new Analytics('1.18', app);
+
+        analytics.init(true, { environment, isDev, instanceId, sessionId, commitId });
+
+        const actionType1 = 'should-report';
+        const actionType2 = 'should-not-report';
```

Comment:
nit: those two action types doesn't make much sense to me. It's not decided by the type whether it should be reported or not, right? One action type would be sufficient here...
```suggestion
        const actionType = 'very-important-action';
```

Reply:
8d6504c701a27900a219bed6b24fa7c3b0f870d6